packages/blocks/src/api/templates.js

```diff
@@ -115,6 +115,23 @@ export function synchronizeBlocksWithTemplate( blocks = [], template ) {
 					normalizedAttributes
 				);
 
+			const hookedBlocksForCurrentBlock = getBlockTypes()?.filter(
+				( { blockHooks } ) => blockHooks && blockName in blockHooks
+			);
+
+			if ( hookedBlocksForCurrentBlock?.length ) {
+				const { metadata, ...otherAttributes } = blockAttributes;
+				blockAttributes = {
+					metadata: {
+						ignoredHookedBlocks: hookedBlocksForCurrentBlock.map(
+							( hookedBlock ) => hookedBlock.name
+						),
+						...blockAttributes.metadata,
+					},
+					...otherAttributes,
+				};
+			}
+
```

Comment:
This is needed for blocks that aren't inserted directly from the inserter, but as inner blocks of a container block. One example would be the Comment Template hook, which is typically inserted as part of the Comments block.

Unfortunately, the [`initialAttributes` concept](https://github.com/WordPress/gutenberg/pull/58553/files#r1474617754) doesn't exist here 😕